### ISSUE TITLE
Breaking Change for file-uploads in v2

### ISSUE THREAD

**Issue by RoiEXLab** [NodeBB/NodeBB#9169]
This is probably a niche issue, but it's inconvenient nevertheless.
[This change](https://github.com/NodeBB/NodeBB/commit/340387c18a273bec333e24c7fda063f4a10e5b0f#diff-e66fce75f5234e3e18d2343810f45677d6ba0f16dc3694d9f08b37461878523b) modified the upload controller to use a different json format for the file upload response.
For v3 this is obviously not a problem (because the route changed as well), but write api v2 uses the same controller, so after updating to 1.15 the code relying on the old format breaks even when still using the v2 route.
Obviously updating the code solves this issue, but pushing the update to our users is always a very slow process so it would be really nice if the existing code would continue to work for a while. Would it be possible to revert back to the old format for v2, or would that be too much of a hassle?

Also I noticed that the upload api isn't listed here for some reason: https://docs.nodebb.org/api/write/#tag/files

**Comment by julianlam** [comment]
Hm, good point, I certainly didn't intend to break the API. I'll see if I can have the old response returned ..

**Comment by djensen47** [comment]
Thanks @julianlam!

**Comment by julianlam** [comment]
@djensen47 @RoiEXLab Are you using `POST /api/v2/util/upload` to trigger this controller?

**Comment by julianlam** [comment]
@djensen47 @RoiEXLab Can you please try with the above commit and let me know if this resolves the issue?

**Comment by djensen47** [comment]
When "installing" a commit or branch, should I use the `git reset --hard`? Or just a plain `git checkout`?

**Comment by julianlam** [comment]
You should be able to cherry-pick this commit straight into whatever branch you are using. `git cherry-pick 23bb9a7`

If it works fine, you can go back to where you were with `git reset --hard HEAD^`

**Comment by djensen47** [comment]
Hmm, the cherry pick failed.

I'm getting this:
```
# git cherry-pick 23bb9a7
.git/hooks/prepare-commit-msg: 35: .git/hooks/prepare-commit-msg: node_modules/run-node/run-node: not found
error: 'prepare-commit-msg' hook failed
fatal: cherry-pick failed
```

**Comment by djensen47** [comment]
Says it failed but it didn't. What's this run-node module issue I'm seeing though?

**Comment by julianlam** [comment]
That would be a custom git hook installed in your `.git/hooks` folder. It's not something we handle (although some of our dev dependencies do)

**Comment by djensen47** [comment]
This fix worked. ✔️

**Comment by julianlam** [comment]
Thank you 👍

**Comment by djensen47** [comment]
> That would be a custom git hook installed in your .git/hooks folder. It's not something we handle (although some of our dev dependencies do)

I don't recall installing a git hook though. This happened after upgrading 1.12 to 1.16. One version at a time, of course, but I can't remember which one started showing the error.

**Comment by julianlam** [comment]
Some git hooks are installed if you run `npm i` or `yarn --dev`.. they're related to linting and such for commits, so that might be what you're seeing.

They are not required for production instances. You can manually go into `.git/hooks` and delete everything that doesn't end in `.sample`

Well, you can delete all the sample files too, it shouldn't make much difference.

**Comment by djensen47** [comment]
Did this fix not make it into 1.16.1?

**Comment by pitaj** [comment]
It won't have the same commit hash. It's here in v1.16.x: https://github.com/NodeBB/NodeBB/commit/171017c38c24fa06fee7333ccf955883e393dfab

**Comment by djensen47** [comment]
This fix is not is definitely not in 1.16.1

**Comment by julianlam** [comment]
Looks like I committed it to the `topic-events` branch instead of `master`, so no, it did not make it in. We'll release 1.16.2 on Wednesday.
